src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -103,18 +107,13 @@ internal class ConnectionManager : IConnectionManager {
             return isRemoved;
         }
 
-        public async Task ConnectAsync(string name, string path, string rCommandLineArguments) {
-            var connection = GetOrCreateConnection(name, path, rCommandLineArguments);
-            await ConnectAsync(connection);
-        }
-
-        public async Task ConnectAsync(IConnection connection) {
+        public async Task ConnectAsync(IConnectionInfo connection) {
             var sessions = _sessionProvider.GetSessions().ToList();
             if (sessions.Any()) {
                 await Task.WhenAll(sessions.Select(s => s.StopHostAsync()));
             }
 
-            if (ActiveConnection != null && ActiveConnection.Id != connection.Id) {
+            if (ActiveConnection != null && (!ActiveConnection.Path.EqualsIgnoreCase(connection.Path)  || _sessionProvider.BrokerUri.IsLoopback)) {
```

Comment:
ok, I've added a `Name` to the `IRHostConnector` in my changes, so I will change this code after merge

---
In reply to: [77671347](https://github.com/Microsoft/RTVS/pull/2246#discussion_r77671347) [](ancestors = 77671347)